### PR TITLE
[GPU] NMSRotated-13

### PR DESCRIPTION
**Pull request by okhovan**
NMSRotated-13 GPU implementation

Ticket CVS-118943

### CODE REVIEW

**Comment by wilson-seok** [comment]
@mitruska Could you please review a fix of reference implementation in src/core/reference/src/op/nms_rotated.cpp? Thank you in advance!

**Review by mitruska — APPROVED** [review]
The changes in reference implementation look good to me, thanks!

File: src/common/transformations/src/transformations/op_conversions/convert_nms_rotated_to_nms_ie_internal.cpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
+// Copyright (C) 2018-2023 Intel Corporation
```
Comment (yeonbok) [inline src/common/transformations/src/transformations/op_conversions/convert_nms_rotated_to_nms_ie_internal.cpp:1]: nit: 2023 only

File: src/plugins/intel_gpu/include/intel_gpu/primitives/non_max_suppression.hpp [hunk 2]
```diff
@@ -17,6 +17,12 @@ namespace cldnn {
 struct non_max_suppression : public primitive_base<non_max_suppression> {
     CLDNN_DECLARE_PRIMITIVE(non_max_suppression)
 
+    enum Rotation {
+        NONE,
+        CLOCKWISE,
+        COUNTERCLOCKWISE
+    };
```
Comment (yeonbok) [inline src/plugins/intel_gpu/include/intel_gpu/primitives/non_max_suppression.hpp:24]: How about share this with ov_ops?
It is using integer in nms_ie_internal.hpp but seems that we can declare one enum and use it from plugin.

File: src/tests/functional/shared_test_classes/src/single_layer/nms_rotated.cpp [hunk 3]
```diff
@@ -0,0 +1,230 @@
+// Copyright (C) 2018-2023 Intel Corporation
```
Comment (yeonbok) [inline src/tests/functional/shared_test_classes/src/single_layer/nms_rotated.cpp:1]: nit:2013

File: src/common/transformations/src/transformations/op_conversions/convert_nms_rotated_to_nms_ie_internal.cpp (hunk 1 above)
Comment (okhovan) [reply src/common/transformations/src/transformations/op_conversions/convert_nms_rotated_to_nms_ie_internal.cpp]: Fixed.

File: src/tests/functional/shared_test_classes/src/single_layer/nms_rotated.cpp (hunk 3 above)
Comment (okhovan) [reply src/tests/functional/shared_test_classes/src/single_layer/nms_rotated.cpp]: Fixed.

File: src/plugins/intel_gpu/include/intel_gpu/primitives/non_max_suppression.hpp (hunk 2 above)
Comment (okhovan) [reply src/plugins/intel_gpu/include/intel_gpu/primitives/non_max_suppression.hpp:24]: I thought of this, but decided not to introduce new enum in ```attr_types.hpp```. This enum is not a part of interface (i.e., it is not used in NMSRotated op - there are boolean ```clockwise``` attribute instead), it is just a detail of implementation of internal operation ```NonMaxSuppressionIEInternal```.

**Review by yeonbok — APPROVED** [review]

**Pull request merged**